pantheon/src/main/java/tech/pegasys/pantheon/cli/RLPSubCommand.java

```diff
@@ -0,0 +1,180 @@
+
+  private final PrintStream out;
+  private final InputStream in;
+
+  @SuppressWarnings("unused")
+  @ParentCommand
+  private PantheonCommand parentCommand;
+
+  @SuppressWarnings("unused")
+  @Spec
+  private CommandSpec spec;
+
+  RLPSubCommand(final PrintStream out, final InputStream in) {
+    this.out = out;
+    this.in = in;
+  }
+
+  @Override
+  public void run() {
+    spec.commandLine().usage(out);
+  }
+
+  /**
+   * RLP encode sub-command
+   *
+   * <p>Encode a JSON data into an RLP hex string.
+   */
+  @Command(
+      name = "encode",
+      description = "This command encodes a JSON typed data into an RLP hex string.",
+      mixinStandardHelpOptions = true)
+  static class EncodeSubCommand implements Runnable {
+
+    @SuppressWarnings("unused")
+    @ParentCommand
+    private RLPSubCommand parentCommand; // Picocli injects reference to parent command
+
+    @SuppressWarnings("unused")
+    @Spec
+    private CommandSpec spec;
+
+    @Option(
+        names = "--from",
+        paramLabel = MANDATORY_FILE_FORMAT_HELP,
+        description = "File containing JSON object to encode",
+        arity = "1..1")
+    private File jsonSourceFile = null;
+
+    @Option(
+        names = "--to",
+        paramLabel = MANDATORY_FILE_FORMAT_HELP,
+        description = "File to write encoded RPL string to.",
+        arity = "1..1")
+    private File rlpTargetFile = null;
+
+    @Override
+    public void run() {
+      checkNotNull(parentCommand);
+      readInput();
+    }
+
+    private void readInput() {
+      // if we have an output file defined, print to it
+      // otherwise print to defined output, usually standard output.
+      StringBuilder jsonData = new StringBuilder();
+
+      if (jsonSourceFile != null) {
+        try {
+          BufferedReader reader = Files.newBufferedReader(jsonSourceFile.toPath(), UTF_8);
+
+          String line;
+          while ((line = reader.readLine()) != null) jsonData.append(line);
+        } catch (IOException e) {
+          throw new ExecutionException(spec.commandLine(), "Unable to read JSON file.");
+        }
+      } else {
+        // get JSON data from standard input
+        try (Scanner scanner = new Scanner(parentCommand.in, UTF_8.name())) {
+          while (scanner.hasNextLine()) {
+            jsonData.append(String.join("", scanner.nextLine().split("\\s")));
+          }
+        }
+      }
+      encode(jsonData.toString());
+    }
+
+    private void encode(final String jsonInput) {
+      // map the json to the object matching the type option
+      // the object must be an RLPEncodable object
+      if (jsonInput == null || jsonInput.isEmpty()) {
+        throw new ExecutionException(
+            spec.commandLine(), "An error occurred while trying to read the JSON data.");
+      } else {
+        try {
+          JsonObject jsonObject = new JsonObject(jsonInput);
+          RLPEncodable objectToEncode = jsonObject.mapTo(IbftExtraData.class);
```

Comment:
Do we make them put in the whole of extra data, or just the bits that are of interest? I.e. for genesis file the round should be 0, the seals should be empty, ideally the validators are ordered.

Reply:
I will change that with the adapter, being able to just use an array.